pkg/parse/parse.go

```diff
@@ -64,29 +64,20 @@ func Parse() {
 		os.Exit(1)
 	}
 
-	currBranch := o.RepoHead.Value()
-
-	cmd := git.Git{Workspace: o.Dir.Value(), Head: currBranch, RepoName: o.RepoName.Value()}
-
-	// TODO: Reintroduce this codepath if we decide the flag finder should be able to clone repos
-	// endpoint := o.CloneEndpoint.Value()
-	// if endpoint != "" {
-	// 	dir, err := ioutil.TempDir("", cmd.RepoName)
-	// 	if err != nil {
-	// 		fatal("Failed to create temp directory for repo clone", err)
-	// 	}
-	// 	defer os.RemoveAll(dir)
-	// 	cmd.Workspace = dir
-	// 	err = cmd.Clone(endpoint)
-	// 	if err != nil {
-	// 		fatal("Unable to clone repo", err)
-	// 	}
-	// }
+	cmd := git.Git{Workspace: o.Dir.Value()}
 
-	headSha, err := cmd.RevParse()
+	currBranch, err := cmd.BranchName()
+	if err != nil {
+		log.Error.Fatalf("error parsing git branch name: %s", err)
+	} else if strings.ToUpper(currBranch) == "HEAD" || currBranch == "" {
+		log.Error.Fatalf("error parsing git branch name: git repo at %s must be checked out to a valid branch", cmd.Workspace)
```

Comment:
This errors out if you actually named your branch `head`, but I don't think that's a huge issue

Reply:
why is a branch name of `head` a problem?

Reply:
It's the output returned by `git rev-parse --abbrev-ref` if you're in a detached state. I suppose we could detect this as a problem earlier on so this check doesn't happen in `parse.go`

Reply:
yeah is there some way to just make sure the repo isn't in a detached state? that being said, i guess this is a nice to have. I could just see a user being surprised that we'd have this restriction. We can also always add support for branches named `head` later w/o a backwards incompatible change